src/scripts/openai-api.js

```diff
@@ -0,0 +1,33 @@
+import OpenAi from 'https://cdn.jsdelivr.net/npm/openai@4.22.1/+esm'
+
+
+
+async function generateResponseApi(actor, input) {
+    const openai = new OpenAi({
+        apiKey: await actor.getFlag("unkenny", "llmAPIKey"),
+        dangerouslyAllowBrowser: true,
+    })
+
+    const chatCompletion = await openai.chat.completions.create({
+        model: await actor.getFlag("unkenny", "model"),
+        messages: [
+            {
+                role: 'system',
+                content: await actor.getFlag("unkenny", "preamble"),
+            },
+            {
+                role: 'user',
+                content: input,
+            }
+        ],
+        temperature: 0,
+        max_tokens: await actor.getFlag("unkenny", "maxNewTokens"),
+        top_p: 1.0,
+        frequency_penalty: 0.0,
```

Comment:
Can GPT's frequency_penalty be related to the repitition_penalty that we store?

Reply:
https://platform.openai.com/docs/api-reference/chat/create#chat-create-frequency_penalty
https://platform.openai.com/docs/api-reference/chat/create#chat-create-presence_penalty

Should be related to the presence_penalty. Maybe make dynamic handlebars for api and local?

Reply:
repetition_penalty is now set as frequency_penalty in ChatGTP.